strawberryfields/tdm/tdmprogram.py

```diff
@@ -294,16 +296,22 @@ def __exit__(self, ex_type, ex_value, ex_tb):
 
         if ex_type is None:
             self.spatial_modes = validate_measurements(self.circuit, self.N)
-            self.construct_circuit()
+            self.timebins = len(self.tdm_params[0])
+            self.total_timebins = self.timebins * self.copies
+            # self.construct_circuit()
+            self.rolled_circuit = self.circuit.copy()
 
-    def construct_circuit(self):
+    def roll(self):
+        """Represent the in a compressed way using without rolling the for loops"""
+        self.circuit = self.rolled_circuit
+
+    def unroll(self):
```

Comment:
Should there be an `self.unrolled_circuit = self.circuit.copy()` at the end of this method? Would `self.unrolled_circuit` be used at all otherwise; will it ever be set?

Reply:
I guess if you never ever pass it to an engine to be run, it is kind of pointless to unroll it.

Reply:
oh, nice catch Theo!